### PR TITLE
Support authentication in DAML triggers

### LINKED ISSUE #3259: Support authentication in DAML triggers

### PR DESCRIPTION
**Pull request by cocreature**
fixes #3259

CHANGELOG_BEGIN

- [DAML Triggers - Experimental] DAML triggers can now be run against
an authenticated ledger.

CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerMain.scala [hunk 1]
```diff
@@ -68,12 +69,20 @@ object RunnerMain {
         val sequencer = new AkkaExecutionSequencerPool("TriggerRunnerPool")(system)
         implicit val ec: ExecutionContext = system.dispatcher
 
+        val tokenHolder = config.accessTokenFile.map(new TokenHolder(_))
+        println(tokenHolder.flatMap(_.token))
```
Comment (stefanobaghino-da) [inline triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerMain.scala]: `println`? Are you sure?

File: triggers/tests/BUILD.bazel [hunk 2]
```diff
@@ -97,6 +98,40 @@ client_server_test(
     server_files = ["$(rootpath :acs.dar)"],
 )
 
+AUTH_TOKEN = "I_CAN_HAZ_AUTH"
```
Comment (stefanobaghino-da) [inline triggers/tests/BUILD.bazel]: 3idero

File: triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerMain.scala (hunk 1 above)
Comment (cocreature) [reply triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/RunnerMain.scala]: Oops, that was just leftover, thanks! Removed it now.

File: triggers/tests/BUILD.bazel (hunk 2 above)
Comment (cocreature) [reply triggers/tests/BUILD.bazel]: Yes you can!

**Pull request merged**